lib/transform.js

```diff
@@ -0,0 +1,104 @@
+var hyperx = require('hyperx')
+
+var EXPR_DELIM = '~!@|@|@!~'
+var CEXPR_DELIM = '~!#|#|#!~'
+
+module.exports.expr = expr
+module.exports.factory = factory
+
+function expr (ex) { return EXPR_DELIM + ex + EXPR_DELIM }
+function unexpr (ex) { return ex.split(EXPR_DELIM)[1] }
+function isexpr (ex) { return typeof ex === 'string' && ex.indexOf(EXPR_DELIM) !== -1 }
+
+function cexpr (cex) { return CEXPR_DELIM + cex + CEXPR_DELIM }
+function uncexpr (cex) { return cex.split(CEXPR_DELIM)[1] }
+function iscexpr (cex) { return typeof cex === 'string' && cex.indexOf(CEXPR_DELIM) !== -1 }
+
+function wrap (value) { return { __wrapper__: value } }
+function unwrap (wrapper) { return wrapper ? wrapper.__wrapper__ : undefined }
+function iswrapper (wrapper) { return typeof wrapper === 'object' && wrapper.__wrapper__ !== undefined }
+
+function concat (a, b) {
+  var aexpr
+  var bexpr
+
+  if (iscexpr(a)) aexpr = a // c-expression
```

Comment:
what's the c in c-expression?

Reply:
c-expression as in concat-expression. They should be probably called meta-expressions since that what they are